app/uk/gov/hmrc/merchandiseinbaggage/service/CalculationService.scala

```diff
@@ -28,9 +28,15 @@ import scala.concurrent.{ExecutionContext, Future}
 import scala.math.BigDecimal.RoundingMode.HALF_UP
 
 @Singleton
-class CalculationService @Inject()(connector: CurrencyConversionConnector)(implicit ec: ExecutionContext) {
+class CalculationService @Inject()(connector: CurrencyConversionConnector, mibConnector: MibConnector)(implicit ec: ExecutionContext) {
   private val logger = Logger("CalculationService")
 
+  def paymentBECalculation(declarationGoods: DeclarationGoods)(implicit hc: HeaderCarrier): Future[PaymentCalculations] =
+    Future.traverse(declarationGoods.goods) { goods =>
+        mibConnector.calculatePayment(goods.calculationRequest).map(result => PaymentCalculation(goods, result))
+      }.map(PaymentCalculations.apply)
+
+  @deprecated("Call backend for calculation. All this code to be removed including CurrencyConversionConnector.")
```

Comment:
currency conversion connector will need to stay as we need to it to display the rates on the payment-calculation page FYI